cloud-agent/service/server/src/main/scala/org/hyperledger/identus/agent/server/jobs/DIDStateSyncBackgroundJobs.scala

```diff
@@ -1,17 +1,53 @@
 package org.hyperledger.identus.agent.server.jobs
 
-import org.hyperledger.identus.agent.walletapi.model.error.GetManagedDIDError
-import org.hyperledger.identus.agent.walletapi.service.ManagedDIDService
-import org.hyperledger.identus.shared.models.WalletAccessContext
+import org.hyperledger.identus.agent.walletapi.service.{ManagedDIDService, WalletManagementService}
+import org.hyperledger.identus.shared.messaging.{Message, MessagingService, Producer}
+import org.hyperledger.identus.shared.models.{WalletAccessContext, WalletAdministrationContext, WalletId}
+import org.hyperledger.identus.shared.utils.DurationOps.toMetricsSeconds
 import zio.*
+import zio.metrics.Metric
 
 object DIDStateSyncBackgroundJobs {
 
-  val syncDIDPublicationStateFromDlt: ZIO[WalletAccessContext with ManagedDIDService, GetManagedDIDError, Unit] =
-    for {
+  private val TOPIC_NAME = "sync-did-state"
+
+  val didPublicationStateSyncTrigger
+      : URIO[ManagedDIDService & WalletManagementService & Producer[WalletId, WalletId], Unit] =
+    ZIO
+      .serviceWithZIO[WalletManagementService](_.listWallets().map(_._1))
```

Comment:
`listWallets` might fail when the number of the wallet is to big to handle in a single batch. Am I right?

Reply:
Yep, this is a "known" limitation of the DID state sync process.